clients/src/main/java/org/apache/kafka/common/network/Selector.java

```diff
@@ -529,6 +581,14 @@ public KafkaChannel channel(String id) {
     }
 
     /**
+     * Return the channel with the specified id if it was disconnected, but not yet closed
+     * since there are outstanding messages to be processed.
+     */
+    public KafkaChannel closingChannel(String id) {
```

Comment:
At the moment you can't get hold of `KafkaChannel` using the `Selectable` interface. If `closingChannel` is added to the interface, perhaps `channel(String id)` needs to be as well? Not sure if you would want one but not the other.